diff --git a/func_8009CA60.c.orig b/func_8009CA60.c
index a8486b8..e7e2e0a 100644
--- a/func_8009CA60.c.orig
+++ b/func_8009CA60.c
@@ -1,9 +1,9 @@
 void func_8009CA60(void *arg0, void *arg1, s32 arg2) {
-    s32 temp_a0;
+    OSIntMask interruptMasks;
 
-    temp_a0 = osSetIntMask(1);
+    interruptMasks = osSetIntMask(OS_IM_NONE);
     arg1->unk4 = arg2;
     arg1->unk0 = (void *) arg0->unk768;
     arg0->unk768 = arg1;
-    osSetIntMask(temp_a0, arg1);
+    osSetIntMask(interruptMasks);
 }
